src/app/components/pool/graphs/heatmap/legend/nodes-heatmap-legend.html

```diff
@@ -5,15 +5,21 @@
         <div class="label">{{item.state}} ({{stateCounter.get(item.state) | async}})</div>
     </bl-clickable>
     <div *ngIf="item.category">
-        <bl-clickable class="legend-item category" (do)="selectState(item.category)" [attr.aria-selected]="item.category === highlightedState" [class.highlighted]="item.category === highlightedState"
+        <bl-clickable class="legend-item category" (do)="selectState(item.category, item.category)" [attr.aria-selected]="item.category === highlightedState" [class.highlighted]="item.category === highlightedState"
```

Comment:
Why are you passing `item.category` twice? The `selectState()` method seems to only accept one argument.

Reply:
I updated selectState() to also take in two params, one of which is called categoryParent which controls the category expansion logic.